cd -P -L

asukaminato0721 opened the pull request:
## Description

Talk about your changes here.

Fixes issue #7206

## TODOs:
<!-- Just check off what what we know been done so far. We can help you with this stuff. -->
- [ ] Changes to fish usage are reflected in user documentation/manpages.
- [ ] Tests have been added for regressions fixed
- [ ] User-visible changes noted in CHANGELOG.rst <!-- Don't document changes for completions inside CHANGELOG.rst, there are lot of such edits -->

asukaminato0721 committed:
Implement cd -L and cd -P

Closes #7206

zanchey committed:
update translations

zanchey committed:
highlight: handle cd now that it has multiple arguments

Work on #11909.

zanchey commented:
Sorry this has taken so long to review. I've updated it for the current codebase, and fixed the highlighting of the new arguments.

One thing I'm not sure about is the long option names. "Physical" is the term other shells use in the documentation, although I don't think it's a particularly good term - there's nothing physical about the file path! "Logical" matches `realpath(1)`, but it means something different there. What about copying `cp(1)` with `--dereference` for `-L` and `--no-dereference` for `-P`?

zanchey committed:
CHANGELOG: log #7206

Closes #11909.

zanchey committed:
fix cd symlink resolution tests on Cygwin

asukaminato0721 commented:
> Sorry this has taken so long to review. I've updated it for the current codebase, and fixed the highlighting of the new arguments.
> 
> One thing I'm not sure about is the long option names. "Physical" is the term other shells use in the documentation, although I don't think it's a particularly good term - there's nothing physical about the file path! "Logical" matches `realpath(1)`, but it means something different there. What about copying `cp(1)` with `--dereference` for `-L` and `--no-dereference` for `-P`?

I agree.

thaJeztah commented:
Thank you both! 🙏
